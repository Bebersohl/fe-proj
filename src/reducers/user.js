export const initialState = {
  creatingUser: false,
  signingInUser: false,
  uid: null,
}

const user = (state = initialState, action) => {
  switch (action.type) {
    case 'CREATE_USER_REQUEST':
      return {
        ...state,
        creatingUser: true
      }
    case 'CREATE_USER_SUCCESS':
      return {
        ...state,
        creatingUser: false,
        uid: action.user.uid
      }
    case 'CREATE_USER_FAIL':
      return {
        ...state,
        creatingUser: false,
        uid: null,
      }
    case 'SIGN_IN_USER_REQUEST':
      return {
        ...state,
        signingInUser: true
      }
    case 'SIGN_IN_USER_SUCCESS':
      return {
        ...state,
        signingInUser: false,
        uid: action.user.uid,
      }
    case 'SIGN_IN_USER_FAIL':
      return {
        ...state,
        signingInUser: false,
        uid: null,
      }
    case 'AUTH_CHANGE':
      return {
        ...state,
        uid: action.user ? action.user.uid : null
      }
    default:
      return state
  }
}

export default user