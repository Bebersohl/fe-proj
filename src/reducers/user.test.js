import user, { initialState } from './user'
const currentUser = {
  uid: 'unique-id',
}

it('handles initial state', () => {
  const next = user(undefined, {})
  expect(next).toMatchSnapshot()
})
it('handles SIGN_IN_USER_SUCCESS', () => {
  const next = user(initialState, {
    type: 'SIGN_IN_USER_SUCCESS',
    user: currentUser,
  })
  expect(next).toMatchSnapshot()
})
it('handles SIGN_IN_USER_FAIL', () => {
  const next = user(initialState, { type: 'SIGN_IN_USER_FAIL' })
  expect(next).toMatchSnapshot()
})
it('handles AUTH_CHANGE with user', () => {
  const next = user(initialState, { type: 'AUTH_CHANGE', user: currentUser })
  expect(next).toMatchSnapshot()
})
it('handles AUTH_CHANGE without user', () => {
  const next = user(initialState, { type: 'AUTH_CHANGE' })
  expect(next).toMatchSnapshot()
})
