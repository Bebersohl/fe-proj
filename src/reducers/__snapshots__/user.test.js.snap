exports[`test handles AUTH_CHANGE with user 1`] = `
Object {
  "creatingUser": false,
  "signingInUser": false,
  "uid": "unique-id",
}
`;

exports[`test handles AUTH_CHANGE without user 1`] = `
Object {
  "creatingUser": false,
  "signingInUser": false,
  "uid": null,
}
`;

exports[`test handles CREATE_USER_FAIL 1`] = `
Object {
  "creatingUser": false,
  "signingInUser": false,
  "uid": null,
}
`;

exports[`test handles CREATE_USER_REQUEST 1`] = `
Object {
  "creatingUser": true,
  "signingInUser": false,
  "uid": null,
}
`;

exports[`test handles CREATE_USER_SUCCESS 1`] = `
Object {
  "creatingUser": false,
  "signingInUser": false,
  "uid": "unique-id",
}
`;

exports[`test handles SIGN_IN_USER_FAIL 1`] = `
Object {
  "creatingUser": false,
  "signingInUser": false,
  "uid": null,
}
`;

exports[`test handles SIGN_IN_USER_REQUEST 1`] = `
Object {
  "creatingUser": false,
  "signingInUser": true,
  "uid": null,
}
`;

exports[`test handles SIGN_IN_USER_SUCCESS 1`] = `
Object {
  "creatingUser": false,
  "signingInUser": false,
  "uid": "unique-id",
}
`;

exports[`test handles initial state 1`] = `
Object {
  "creatingUser": false,
  "signingInUser": false,
  "uid": null,
}
`;
